tests/integration/security/sds_ingress/util/util.go

```diff
@@ -644,3 +644,32 @@ func SetInstances(apps echo.Services) error {
 	}
 	return nil
 }
+
+// Some cloud platform may throw the following error during creation of the service with mixed TCP/UDP protocols:
+// "Error syncing load balancer: failed to ensure load balancer: mixed protocol is not supported for LoadBalancer".
+// Make sure the service is up and running before proceeding with the test.
+func UntilIngressQUICSvcReady(t framework.TestContext, ns string) error {
+	return retry.UntilSuccess(func() error {
+		services, err := t.Clusters().Default().Kube().CoreV1().Services(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+		if len(services.Items) == 0 {
+			return fmt.Errorf("still waiting the service in namespace to be created")
+		}
+
+		// Verify that services are not stuck with creation error
+		for _, s := range services.Items {
+			events, err := t.Clusters().Default().Kube().CoreV1().Events(ns).List(context.TODO(), metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
```

Comment:
It should be sufficient to fetch the events only once instead of doing it in the loop. So, we can move this to Line 660.

Reply:
Got it.

Changed.